src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -104,7 +104,16 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     }
 
     public canAutoConnect(): boolean {
-        return !hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.MFA_SERIAL) && !this.isSsoProfile()
+        // check if SSO token is still valid
+        if (this.isSsoProfile()) {
+            const ssoCredentialProvider = this.makeSsoProvider()
+            return ssoCredentialProvider.ssoAccessTokenProvider.cache.loadAccessToken(
+                this.profile[SHARED_CREDENTIAL_PROPERTIES.SSO_START_URL]!
+            ) === undefined
+                ? false
+                : true
```

Comment:
Nits:
* Is it worth turning this into a public function on the ssoCredentialProvider instead of exposing the cache?
* This can condense to
```
            return !!ssoCredentialProvider.ssoAccessTokenProvider.cache.loadAccessToken(
                this.profile[SHARED_CREDENTIAL_PROPERTIES.SSO_START_URL]!
            )

Reply:
Looking over this again, I don't think we even need the ssoCredentialProvier.  We just need the a `DiskCache` and the `start_url` found in the profile.  I'm thinking I can leave all of those private and condense this greatly.